dht_bootstrap.go

```diff
@@ -77,7 +77,24 @@ func (dht *IpfsDHT) BootstrapWithConfig(cfg BootstrapConfig) (goprocess.Process,
 		return nil, fmt.Errorf("invalid number of queries: %d", cfg.Queries)
 	}
 
-	proc := periodicproc.Tick(cfg.Period, dht.bootstrapWorker(cfg))
+	tickch := make(chan struct{}, 1)
+	proc := periodicproc.OnSignal(tickch, dht.bootstrapWorker(cfg))
```

Comment:
So, OnSignal is useful when you already have a channel but, in this case, I'd just fire off your own goroutine:

```go
dht.Process().Go(func(p goprocess.Process) {
    for {
       select {
       ...
       case <-p.Closing():
           return
       }
    }
})
```

Reply:
ah, totally!